lib/profanity.js

```diff
@@ -18,22 +18,49 @@ var DEFAULT_REPLACEMENTS = [
         'rainbows',
         'unicorn'
     ],
-    DEFAULT_REGEX = getListRegex(swearwords);
+    DEFAULT_PATTERN = listToPattern(swearwords);
 
-function getListRegex (list) {
-  // we want to treat all characters in the word as literals, not as regex specials (e.g. shi+)
-  function escapeRegexChars(word) { return word.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&'); };
-  return new RegExp('\\b(' + list.map(escapeRegexChars).join('|') + ')\\b', 'gi');
+function listToPattern(list) {
+    // we want to treat all characters in the word as literals, not as regex specials (e.g. shi+)
+    function escapeRegexChars(word) { return word.replace(/[-\/\\^$*+?.()|[\]{}]/g, '\\$&'); };
+
+    return '(' + list.map(escapeRegexChars).join('|') + ')';
+}
+
+function getListRegex(options) {
+    var pattern = DEFAULT_PATTERN;
+
+    options = options || {};
+
+    if (options.forbiddenList) {
+        pattern = listToPattern(options.forbiddenList);
+    }
+
+    if (!options.substrings) {
+        pattern = '\\b' + pattern + '\\b';
+    }
+
+    return new RegExp(pattern, 'gi');
 }
 
-function check (target, forbiddenList) {
+function check(target, options) {
     var targets = [],
-        regex = forbiddenList ? getListRegex(forbiddenList) : DEFAULT_REGEX;
+        fRef,
+        regex;
+
+    options = options || {};
+
+    // Backwards compatibility
+    if (Array.isArray(options)) {
```

Comment:
Do you think it could be a good idea to print a deprecation message here when users give an array instead of the options object?